File: workspace/platf-bizz-prjs/drr-tnt-swfabrik-europe/food-market-01-service/src/test/java/com/darueira/foodmarket/service01/FoodTradingDomainTest.java
package com.darueira.foodmarket.service01;

import com.darueira.foodmarket.service01.domain.model.FoodTrading;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;

import static org.junit.jupiter.api.Assertions.*;

class FoodTradingDomainTest {

    @Test
    @DisplayName("Should validate and compute total price for food trading")
    void testFoodTradingValidationAndCalculation() {
        FoodTrading trading = FoodTrading.builder()
                .itemName("Valencia Oranges")
                .quantity(new BigDecimal("100.00"))
                .unitPrice(new BigDecimal("2.50"))
                .traderName("Iberia Fresh SL")
                .build();

        trading.validateAndCalculate();

        assertNotNull(trading.getTradingId());
        assertTrue(trading.getTradingId().startsWith("TRD-JAVA-"));
        assertEquals(new BigDecimal("250.0000"), trading.getTotalPrice());
        assertEquals("CONFIRMED", trading.getStatus());
        assertNotNull(trading.getCreatedAt());
    }

    @Test
    @DisplayName("Should reject invalid quantity")
    void testInvalidQuantity() {
        FoodTrading trading = FoodTrading.builder()
                .itemName("Valencia Oranges")
                .quantity(new BigDecimal("-5.00"))
                .unitPrice(new BigDecimal("2.50"))
                .traderName("Iberia Fresh SL")
                .build();

        assertThrows(IllegalArgumentException.class, trading::validateAndCalculate);
    }
}

File: workspace/platf-bizz-prjs/drr-tnt-swfabrik-europe/food-market-02-service/src/test/kotlin/com/darueira/foodmarket/service02/FoodTradingDomainTest.kt
package com.darueira.foodmarket.service02

import com.darueira.foodmarket.service02.domain.model.FoodTrading
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import java.math.BigDecimal

class FoodTradingDomainTest {

    @Test
    @DisplayName("Should validate and compute total price in Kotlin")
    fun testKotlinFoodTradingValidation() {
        val trading = FoodTrading(
            marketId = "MKT-EU-02-QUARKUS",
            itemName = "Greek Kalamata Olives",
            quantity = BigDecimal("200.00"),
            unitPrice = BigDecimal("8.50"),
            traderName = "Hellenic Goods SA"
        )
        trading.validateAndCalculate()

        assertNotNull(trading.tradingId)
        assertTrue(trading.tradingId!!.startsWith("TRD-KT-"))
        assertEquals(BigDecimal("1700.0000"), trading.totalPrice)
        assertEquals("CONFIRMED", trading.status)
    }

    @Test
    @DisplayName("Should reject invalid blank name in Kotlin")
    fun testInvalidBlankName() {
        val trading = FoodTrading(
            marketId = "MKT-EU-02-QUARKUS",
            itemName = "",
            quantity = BigDecimal("200.00"),
            unitPrice = BigDecimal("8.50"),
            traderName = "Hellenic Goods SA"
        )
        assertThrows(IllegalArgumentException::class.java) {
            trading.validateAndCalculate()
        }
    }
}
